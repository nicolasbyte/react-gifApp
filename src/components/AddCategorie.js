import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategorie = ( { setCategories } ) => {

    const [categorie, setCategorie] = useState('');

    const addHandle = ( e ) => {
        
        setCategorie( e.target.value )
    }

    const addSumbit = ( e ) => {
        e.preventDefault();

        if ( categorie.trim().length > 2) {
            
            setCategories( cat => [ categorie , ...cat ] );
            setCategorie( '' );
        }


    }
    return (
        
            <form onSubmit = { addSumbit } >
                <p> { categorie } </p>
                <input
                        type     = "text"
                        value    = { categorie }
                        onChange = { addHandle }
                    />
            </form>

    );

};


AddCategorie.propTypes = {

    setCategories : PropTypes.func.isRequired 
}
