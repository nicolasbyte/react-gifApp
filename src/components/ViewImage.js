import React from 'react';
import PropTypes from 'prop-types';


export const ViewImage = ( { title , url } ) => {
    return (
        <div className = "card animate__animated animate__fadeIn">
                <img src = { url } alt = { title } />
                <p> { title } </p>           
        </div>
    );
};


ViewImage.propTypes = {
    title : PropTypes.string.isRequired,
    url   : PropTypes.string.isRequired
};
