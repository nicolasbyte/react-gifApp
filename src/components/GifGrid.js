import React from 'react';
import PropTypes from 'prop-types';

import { useGifFetch } from '../hooks/useGifFetch';
import { ViewImage } from './ViewImage';

export const GifGrid =  ( { category } ) => {

    
    const { loading , data : images } = useGifFetch( category );



    
    

    return (
        <>
        <h3> { category } </h3>
        { loading && <p>Cargando...</p> }
        <div className = "card-grid animate__animated animate__fadeIn">
            
                {
                    images.map(  img  => (
                        <ViewImage 
                                    key = { img.id }
                                    { ...img } />
                    ) )
                }
            
        </div>
        </>
    );
}


GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}