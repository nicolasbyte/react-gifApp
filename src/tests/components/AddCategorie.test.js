import { shallow } from "enzyme";
import { AddCategorie } from "../../components/AddCategorie";


describe('Este es el test del componente del AddCategorie', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategorie  setCategories = { setCategories }/>);
    
    beforeEach( () => {
        
        jest.clearAllMocks();
            wrapper = shallow( <AddCategorie  setCategories = { setCategories }/>);

    })
    test('imagen del AddCAtegorie', () => {    

        expect( wrapper ).toMatchSnapshot();


    });

    test('debe cambiar la caja de texto', () => {

        const value = 'Hola Mundo';
        wrapper.find('input').simulate('change', { target : { value } }     );
        
        expect( wrapper.find('p').text().trim() ).toBe( value ); 
        
    });
    
    test('no debe dispararse la funciÃ³n setCategories', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        
        expect( setCategories ).not.toHaveBeenCalled();
    });
    
    
    test('debe llamarse el setCategories y el input ser igual a string vacio', () => {
        
        const value = 'Hola Mundo';
        wrapper.find('input').simulate('change', { target : { value } }     );
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        
        
        
        expect( setCategories).toHaveBeenCalled();
        
        expect( wrapper.find('input').prop('value').trim() ).toBe( '' );
          
    });
    
    
    
    
});
