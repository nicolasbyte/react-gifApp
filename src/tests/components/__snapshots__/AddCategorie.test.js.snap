// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Este es el test del componente del AddCategorie imagen del AddCAtegorie 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
